terminal.cpp

```diff
@@ -213,18 +219,111 @@ void Terminal::keyPress(int key, int modifiers)
+                  (modifiers & Qt::ControlModifier ? 4 : 0);
+
+    if (modcode == 0) {
+        char cursorModif='[';
+        if(iAppCursorKeys)
+            cursorModif = 'O';
+
+        if( key==Qt::Key_Up )
+            toWrite += QString("%1%2A").arg(ch_ESC).arg(cursorModif).toLatin1();
+        if( key==Qt::Key_Down )
+            toWrite += QString("%1%2B").arg(ch_ESC).arg(cursorModif).toLatin1();
+        if( key==Qt::Key_Right )
+            toWrite += QString("%1%2C").arg(ch_ESC).arg(cursorModif).toLatin1();
+        if( key==Qt::Key_Left )
+            toWrite += QString("%1%2D").arg(ch_ESC).arg(cursorModif).toLatin1();
+        if( key==Qt::Key_PageUp )
+            toWrite += QString("%1[5~").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_PageDown )
+            toWrite += QString("%1[6~").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_Home )
+            toWrite += QString("%1OH").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_End )
+            toWrite += QString("%1OF").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_Insert )
+            toWrite += QString("%1[2~").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_Delete )
+            toWrite += QString("%1[3~").arg(ch_ESC).toLatin1();
+
+        if( key==Qt::Key_F1 )
+            toWrite += QString("%1OP").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F2 )
+            toWrite += QString("%1OQ").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F3 )
+            toWrite += QString("%1OR").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F4 )
+            toWrite += QString("%1OS").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F5 )
+            toWrite += QString("%1[15~").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F6 )
+            toWrite += QString("%1[17~").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F7 )
+            toWrite += QString("%1[18~").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F8 )
+            toWrite += QString("%1[19~").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F9 )
+            toWrite += QString("%1[20~").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F10 )
+            toWrite += QString("%1[21~").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F11 )
+            toWrite += QString("%1[23~").arg(ch_ESC).toLatin1();
+        if( key==Qt::Key_F12 )
+            toWrite += QString("%1[24~").arg(ch_ESC).toLatin1();
 
-    if( key==Qt::Key_Up )
-        toWrite += QString("%1%2A").arg(ch_ESC).arg(cursorModif).toLatin1();
-    if( key==Qt::Key_Down )
-        toWrite += QString("%1%2B").arg(ch_ESC).arg(cursorModif).toLatin1();
-    if( key==Qt::Key_Right )
-        toWrite += QString("%1%2C").arg(ch_ESC).arg(cursorModif).toLatin1();
-    if( key==Qt::Key_Left )
-        toWrite += QString("%1%2D").arg(ch_ESC).arg(cursorModif).toLatin1();
+    } else {
+        char modChar = '1' + modcode;
+
+        if( key==Qt::Key_Up )
+            toWrite += QString("%1[1;%2A").arg(ch_ESC).arg(modChar).toLatin1();
```

Comment:
You'd also need to turn off the effect of iAppCursorKeys (thats a terminal mode) for the case of the modifiers, and also
note not all of them are just normal with ;m (m=modifier) tacked in the middle, eg F1-F4, home/end (those were the first ones i noticed from the diff) are minus 'O' add '[1;m'  for normal vs with modifiers.

FYI: My quickly googled reference on the codes was https://code.google.com/p/mintty/wiki/Keycodes
So basically the splitting of modifiers/without was caused by following the table styles there (seperate codes for plain and modifiers) and i wanted to make sure i didnt break anything, but i know it could be more compact/smarter.
I'm not sure if it'd be actually of any real use to do that though.